apps/api/src/app/integrations/usecases/get-active-integration/get-active-integration.usecase.ts

```diff
@@ -1,20 +1,108 @@
-    return await this.getDecryptedIntegrationsUsecase.execute(
+  async execute(command: GetActiveIntegrationsCommand): Promise<GetActiveIntegrationResponseDto[]> {
+    const activeIntegration = await this.getDecryptedIntegrationsUsecase.execute(
       GetDecryptedIntegrationsCommand.create({
         organizationId: command.organizationId,
-        active: true,
         userId: command.userId,
+        active: true,
       })
     );
+
+    if (!activeIntegration.length) {
+      return [];
+    }
+
+    const activeIntegrationChannelTypes = this.getDistinctChannelTypes(activeIntegration);
+    const selectedIntegrations = await this.getSelectedIntegrations(command, activeIntegrationChannelTypes);
+
+    return this.mapBySelectedIntegration(activeIntegration, selectedIntegrations);
+  }
+
+  private getDistinctChannelTypes(activeIntegration: IntegrationEntity[]): ChannelTypeEnum[] {
+    return activeIntegration.map((integration) => integration.channel).filter(this.distinct);
+  }
+
+  distinct = (value, index, self) => {
+    return self.indexOf(value) === index;
+  };
+
+  private mapBySelectedIntegration(
+    activeIntegration: IntegrationEntity[],
+    selectedIntegrations: IntegrationEntity[]
+  ): GetActiveIntegrationResponseDto[] {
+    return activeIntegration.map((integration) => {
+      const selected = selectedIntegrations.find((selectedIntegration) => selectedIntegration._id === integration._id);
+
+      return selected ? { ...integration, selected: true } : { ...integration, selected: false };
+    });
+  }
+
+  private async getSelectedIntegrations(
+    command: GetActiveIntegrationsCommand,
+    activeIntegrationChannelTypes: ChannelTypeEnum[]
+  ) {
+    const environments = await this.environmentRepository.findOrganizationEnvironments(command.organizationId);
+
+    const integrationPromises = this.selectIntegrationByEnvironment(
+      environments,
+      command,
+      activeIntegrationChannelTypes
+    );
+
+    return (await Promise.all(integrationPromises)).filter(notNullish);
+  }
+
+  private selectIntegrationByEnvironment(
+    environments,
+    command: GetActiveIntegrationsCommand,
+    activeIntegrationChannelTypes: ChannelTypeEnum[]
+  ) {
+    return environments.flatMap((environment) =>
+      this.selectIntegrationByChannelType(environment._id, command, activeIntegrationChannelTypes)
+    );
+  }
+
+  private selectIntegrationByChannelType(
+    environmentId,
+    command: GetActiveIntegrationsCommand,
+    activeIntegrationChannelTypes: ChannelTypeEnum[]
+  ) {
+    return activeIntegrationChannelTypes.map(async (channelType) => {
+      return await this.selectIntegration.execute(
+        SelectIntegrationCommand.create({
+          environmentId: environmentId,
+          organizationId: command.organizationId,
+          userId: command.userId,
+          channelType: channelType as ChannelTypeEnum,
+          providerId: command.providerId,
+        })
+      );
+    });
```

Comment:
I would combine all these in one function, it's harder to read if they are that split granually

Reply:
I kindly disagree with this because from a perspective of functional programming. Maybe one for a tech summit to discuss style guide and agreements we would like to have regarding style. What do you think?

Reply:
initially, it was one function while selectIntegrationByEnvironment was one inline nested functions but I was not sure it will be readable for someone seeing it for the first time.